core/templates/pages/exploration-editor-page/exploration-editor-page.component.spec.ts

```diff
@@ -0,0 +1,636 @@
+          .toHaveBeenCalledWith(explorationId);
+        expect(RouterService.navigateToMainTab).toHaveBeenCalled();
+        $timeout.flush();
+
+        expect(ctrl.tutorialInProgress).toBe(true);
+
+        ctrl.onFinishTutorial();
+        expect(SiteAnalyticsService.registerFinishTutorialEvent)
+          .toHaveBeenCalledWith(explorationId);
+        expect(ctrl.tutorialInProgress).toBe(false);
+      });
+
+      it('should check if improvements tab is enabled', function() {
+        var isInitializedSpy = spyOn(
+          ExplorationFeaturesService, 'isInitialized');
+        var isImprovementsTabEnabledSpy = spyOn(
+          ExplorationFeaturesService, 'isImprovementsTabEnabled');
+        isInitializedSpy.and.returnValue(true);
+        isImprovementsTabEnabledSpy.and.returnValue(true);
+        expect(ctrl.isImprovementsTabEnabled()).toBe(true);
+
+        isInitializedSpy.and.returnValue(false);
+        isImprovementsTabEnabledSpy.and.returnValue(true);
+        expect(ctrl.isImprovementsTabEnabled()).toBe(false);
+
+        isInitializedSpy.and.returnValue(true);
+        isImprovementsTabEnabledSpy.and.returnValue(false);
+        expect(ctrl.isImprovementsTabEnabled()).toBe(false);
+
+        isInitializedSpy.and.returnValue(false);
+        isImprovementsTabEnabledSpy.and.returnValue(false);
+        expect(ctrl.isImprovementsTabEnabled()).toBe(false);
+      });
+
+      it('should decline tutorial when dismissing welcome exploration modal',
+        function() {
+          spyOn($uibModal, 'open').and.returnValue({
+            result: $q.reject(explorationId)
+          });
+
+          expect(ctrl.tutorialInProgress).toBe(false);
+
+          ctrl.showWelcomeExplorationModal();
+          $scope.$apply();
+
+          expect(SiteAnalyticsService.registerDeclineTutorialModalEvent)
+            .toHaveBeenCalled();
+          expect(ctrl.tutorialInProgress).toBe(false);
+        });
+
+      it('should toggle exploration warning visibility', function() {
+        expect(ctrl.areExplorationWarningsVisible).toBe(false);
+
+        ctrl.toggleExplorationWarningVisibility();
+        expect(ctrl.areExplorationWarningsVisible).toBe(true);
+
+        ctrl.toggleExplorationWarningVisibility();
+        expect(ctrl.areExplorationWarningsVisible).toBe(false);
+      });
+
+      it('should get exploration url', function() {
+        expect(ctrl.getExplorationUrl(explorationId)).toBe('/explore/exp1');
+        expect(ctrl.getExplorationUrl()).toBe('');
+      });
+
+      it('should get active tab name', function() {
+        var activeTabNameSpy = spyOn(RouterService, 'getActiveTabName');
+
+        activeTabNameSpy.and.returnValue('preview');
+        expect(ctrl.getActiveTabName(activeTabNameSpy)).toBe('preview');
+
+        activeTabNameSpy.and.returnValue('history');
+        expect(ctrl.getActiveTabName(activeTabNameSpy)).toBe('history');
+      });
+
+      it('should change element scroll top when calling fn on index 1 of' +
```

Comment:
I don't really understand this test and all the ones below it...

Reply:
It's calling the function declared on `fn` property of the respective indexes in `ctrl.EDITOR_TUTORIAL_OPTIONS` and it's asserting if `element.animate` is being called.